### PR TITLE
Dockerize

### LINKED ISSUE #1: Dockerize the Stream Registry
Desired Behavior
==============================
First time users especially if they are developers who want to play with this offering would find it useful to have a Dockerized install available to them. Provide some sample data, topology and prod/consumers out of the box so there is a script to follow. Make this script become available in the documentation.

Benefits
==============================

* Make installation and on-boarding easier
* A simple working example out of the box to help educate and get value
* Documentation of key customer journey touch points

### PR DESCRIPTION
**Pull request by cricket007**
# stream-registry PR

Closes #1 

### Added
* `jib-maven-plugin`
* `docker-compose.yml`

### Changed
* README updated

# PR Checklist Forms

- [x] CHANGELOG.md updated
- [x] Reviewer assigned
- [x] PR assigned (presumably to submitter)
- [x] Labels added (enhancement, bug, documentation)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,18 +47,38 @@ See the [architecture/northstar documentation](https://homeaway.github.io/stream
 
 ## Building locally
 
+Stream Registry is built using Maven. For convenience, we have wrapped each Maven command in a [`Makefile`](Makefile).
+If you do not have `make` installed, please consult this file for build commands.
+
+Stream Registry can either be deployed as a standalone JAR file, or as a Docker image.
```
[the remark this answers is not in the document]
Comment (cricket007) [reply README.md]: Done

File: README.md [hunk 2]
```diff
@@ -72,7 +94,7 @@ Stream Registry can then be started
 make run
 ```
 
-Check that the application's Swagger API is running at http://localhost:8080/swagger
+Once Stream Registry has started, check that the application's Swagger API is running at http://localhost:8080/swagger
```
Comment (neoword) [inline README.md]: should this be `http://localhost:8080/private/swagger` ?

File: assembly/src/main/resources/stream-registry-config.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  title: Stream Registry API
+  version: v0
+  description: Stream Registry API
+  contact: stream-registry-committers@groups.homeawaycorp.com
+  scan: true
+
+httpClient:
+  timeout: 10000ms
+  connectionTimeout: 10000ms
+  connectionRequestTimeout: 10000ms
```
Comment (neoword) [inline assembly/src/main/resources/stream-registry-config.yaml]: Why do we need this config file built as part of the assembly (src/main/resources)?
Maybe put this in a separate section (not in `src/main/resources` but maybe in `src/main/docker`) ?

File: assembly/pom.xml [hunk 4]
```diff
@@ -46,6 +50,61 @@
...
+                    <plugin>
+                        <groupId>com.google.cloud.tools</groupId>
+                        <artifactId>jib-maven-plugin</artifactId>
+                        <configuration>
+                            <to>
+                                <image>homeaway/stream-registry</image>
+                                <tags>
+                                    <tag>${project.version}</tag>
+                                    <tag>latest</tag>
+                                </tags>
```
Comment (neoword) [inline assembly/pom.xml]: VERY COOL!
❤️

**Comment by neoword** [comment]
Please update CHANGELOG and target this for 0.4.1

File: README.md (hunk 2 above)
Comment (cricket007) [reply README.md]: Nope

```
[stream-registry_1  | [19:37:18.353] [main] INFO   i.d.j.DropwizardResourceConfig -  The following paths were found for the configured resources:
[stream-registry_1  | 
[stream-registry_1  |      GET     /swagger (io.federecio.dropwizard.swagger.SwaggerResource)
```

File: assembly/src/main/resources/stream-registry-config.yaml (hunk 3 above)
Comment (cricket007) [reply assembly/src/main/resources/stream-registry-config.yaml]: Jib will pull from `src/main/resources` or `src/main/jib`. 

I need the file to be put into the container somewhere... The one problem I have with this approach is that I don't like having two config files, and Jib can't find the file from the root of the project.

Comment (neoword) [reply assembly/src/main/resources/stream-registry-config.yaml]: why not `src/main/jib` ?

**Pull request merged**